src/jobs/deploy.yml

```diff
@@ -35,6 +39,7 @@ steps:
   - setup_remote_docker:
       version: <<parameters.docker-version>>
       docker_layer_caching: true
+  - steps: <<parameters.pre-deploy>>
```

Comment:
This should come after the packaging.

Otherwise you might end up with the pre deploy steps running (read only mode) and be left in that state if the packaging fails.